includes/enrolment/class-sensei-enrolment-provider-state.php

```diff
@@ -62,12 +62,26 @@ public static function from_serialized_array( Sensei_Enrolment_Provider_State_St
 			return false;
 		}
 
-		$provider_data = isset( $data['d'] ) ? array_filter( array_map( [ __CLASS__, 'sanitize_data' ], $data['d'] ) ) : [];
-		$logs          = isset( $data['l'] ) ? array_filter( array_map( [ __CLASS__, 'sanitize_logs' ], $data['l'] ) ) : [];
+		$provider_data = isset( $data['d'] ) ? array_map( [ __CLASS__, 'sanitize_data' ], $data['d'] ) : [];
```

Comment:
Why is this block of code needed?

Reply:
For restoring the object from its JSON representation. Do you mean why are we sanitizing? Partly because this is a pretty generic implementation and we can't rule out that it could be used to store some kind of user input. I've been sanitizing what comes from stored JSON that might have user input.